app/scripts/modules/oracle/serverGroup/details/serverGroupDetails.controller.js

```diff
@@ -0,0 +1,216 @@
+        controller: 'oraclebmcsRollbackServerGroupCtrl as ctrl',
+        resolve: {
+          serverGroup: () => this.serverGroup,
+          disabledServerGroups: () => {
+            let sgSummary = _.find(app.serverGroups.data, {
+              name: this.serverGroup.name,
+              account: this.serverGroup.account,
+              region: this.serverGroup.region
+            });
+            let cluster = _.find(app.clusters, {name: sgSummary.cluster, account: this.serverGroup.account});
+            return _.filter(cluster.serverGroups, {isDisabled: true, region: this.serverGroup.region});
+          },
+          application: () => app
+        }
+      });
+    };
+
+    this.disableServerGroup = () => {
+      let serverGroup = this.serverGroup;
+
+      let taskMonitor = {
+        application: app,
+        title: 'Disabling ' + serverGroup.name
+      };
+
+      let submitMethod = (params) => serverGroupWriter.disableServerGroup(serverGroup, app, params);
+
+      let confirmationModalParams = {
+        header: 'Really disable ' + serverGroup.name + '?',
+        buttonText: 'Disable ' + serverGroup.name,
+        account: serverGroup.account,
+        taskMonitorConfig: taskMonitor,
+        platformHealthOnlyShowOverride: app.attributes.platformHealthOnlyShowOverride,
+        platformHealthType: 'Oracle',
+        submitMethod: submitMethod,
+        askForReason: true,
+      };
+
+      serverGroupWarningMessageService.addDisableWarningMessage(app, serverGroup, confirmationModalParams);
+
+      if (app.attributes.platformHealthOnlyShowOverride && app.attributes.platformHealthOnly) {
+        confirmationModalParams.interestingHealthProviderNames = ['Oracle'];
+      }
+
+      confirmationModalService.confirm(confirmationModalParams);
+    };
+
+    this.enableServerGroup = () => {
+      let serverGroup = this.serverGroup;
+
+      let taskMonitor = {
+        application: app,
+        title: 'Enabling ' + serverGroup.name,
+      };
+
+      let submitMethod = (params) => serverGroupWriter.enableServerGroup(serverGroup, app, params);
+
+      let confirmationModalParams = {
+        header: 'Really enable ' + serverGroup.name + '?',
+        buttonText: 'Enable ' + serverGroup.name,
+        account: serverGroup.account,
+        taskMonitorConfig: taskMonitor,
+        platformHealthOnlyShowOverride: app.attributes.platformHealthOnlyShowOverride,
+        platformHealthType: 'Oracle',
+        submitMethod: submitMethod,
+        askForReason: true,
+      };
+
+      if (app.attributes.platformHealthOnlyShowOverride && app.attributes.platformHealthOnly) {
+        confirmationModalParams.interestingHealthProviderNames = ['Oracle'];
+      }
+
+      confirmationModalService.confirm(confirmationModalParams);
+    };
+
+    let cancelLoader = () => {
+      this.state.loading = false;
+    };
+
+    retrieveServerGroup().then(() => {
+      cancelLoader();
```

Comment:
Based on how `retrieveServerGroup` is written, `cancelLoader` will be called before `retrieveNetwork`, `retrieveImage`, and `retrieveSubnet` resolve. That's okay, but might be hard to reason about (and things might suddenly pop in on the details panel).